connector/connect/common/src/main/protobuf/spark/connect/relations.proto

```diff
@@ -76,6 +76,8 @@ message Relation {
     StatFreqItems freq_items = 106;
     StatSampleBy sample_by = 107;
 
+    int64 server_side_dataframe_id = 108;
```

Comment:
This should be in `RelationCommon`?

Reply:
This should be it's own proper message.

```

ServerSideDataFrame server_side_data_frame = 108;

message ServerSideDataFrame {
   int64 id = 0;
}
```

Reply:
It makes sense to have its own message if it is a rel type.

Reply:
@amaliujia 

> This should be in RelationCommon?

No. Because this might be for either DataFrame object or other type objects (e.g. Model type object)

Reply:
Can we use `RelationCommon` ? I am not quite familiar with Relation part,
the server side dataframe might contain some stage (e.g. using RDD api) that Spark connect proto does not support

Reply:
The problem here is that I do not fully understand the semantics of this `ServerSideDataFrame`. If you can help us understand what exactly `ServerSideDataFrame` will do, probably we can help to see how does it better fit in the Connect proto.

Reply:
@amaliujia 

You can see this line:

https://github.com/apache/spark/blob/a7a5e30a2db8561ebd51f90e3690ac5f75c6ae60/connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala#L119

```
class SparkConnectPlanner(val sessionHolder: SessionHolder) {
  ...
  def transformRelation(rel: proto.Relation): LogicalPlan = {
    rel.getRelTypeCase match {
      ...
      case proto.Relation.RelTypeCase.SERVER_SIDE_DATAFRAME_ID =>
        sessionHolder.serverSideObjectManager.getObject(rel.getServerSideDataframeId)
          .asInstanceOf[Dataset[Any]].logicalPlan
      ...
  }
```
i.e., it means we get the spark dataframe plan via query the id from `sessionHolder.serverSideObjectManager`

Reply:
what is the dataframe/logicalplan here? is it the output of model prediction (`model.transform`) ?

Reply:
> is it the output of model prediction (model.transform)

Yes. This is one case.

Other cases are: Some model attributes and model summary methods might return spark DataFrame.